### PR TITLE
Remove buffer gem and related code: step 1

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Refactor

## Description

This code removes all functionality related to Buffer. Buffer stopped supporting the API we used a while ago, so this was absolute legacy code.

Note: columns and tables will be removed in a follow-up, in this PR they have only been added to `ignored_columns` on the relevant models.

## Related Tickets & Documents

https://github.com/forem/internalEngineering/issues/397

## QA Instructions, Screenshots, Recordings

* All specs should pass
* Click around the articles admin area and make sure things are working
* Click around articles and make sure things are working
 
### UI accessibility concerns?

n/a

## Added tests?

- [X] No, and this is why: only removing code 

## [Forem core team only] How will this change be communicated?

- [X] I will share this change internally with the appropriate teams

### CODE REVIEW

**Review by rhymes — CHANGES_REQUESTED** [review]
Don't know if you plan to remove them in the following PR when you remove the columns but I found a few other files containing pieces of Buffer:

https://github.com/forem/forem/blob/18a0fd5f61dd545655c72e2a4ecb7dbedcdbb001/spec/rails_helper.rb#L166-L167

https://github.com/forem/forem/blob/master/spec/factories/buffer_updates.rb

https://github.com/forem/forem/blob/master/app/policies/buffer_update_policy.rb and its spec

https://github.com/forem/forem/blob/master/app/controllers/buffer_updates_controller.rb

https://github.com/forem/forem/blob/master/app/controllers/admin/buffer_updates_controller.rb

https://github.com/forem/forem/blob/master/app/javascript/admin/controllers/buffer_controller.js and https://github.com/forem/forem/blob/master/app/javascript/admin/__tests__/controllers/buffer_controller.test.js

https://github.com/forem/forem/blob/18a0fd5f61dd545655c72e2a4ecb7dbedcdbb001/app/lib/constants/role.rb#L20

https://github.com/forem/forem/blob/18a0fd5f61dd545655c72e2a4ecb7dbedcdbb001/app/models/tag.rb#L69-L73

plus the `BufferUpdate` model but that's step 2, correct?

File: lib/data_update_scripts/20200825095213_remove_orphaned_buffer_updates_by_article.rb [hunk 1]
```diff
@@ -1,13 +1,7 @@
...
   class RemoveOrphanedBufferUpdatesByArticle
     def run
-      # Delete all BufferUpdates belonging to Articles that don't exist anymore
-      ActiveRecord::Base.connection.execute(
-        <<~SQL.squish,
-          DELETE FROM buffer_updates
-          WHERE article_id NOT IN (SELECT id FROM articles);
-        SQL
-      )
+      # We removed Buffer updates from the codebase, so this is now a no-op.
```
Comment (jgaskins) [inline lib/data_update_scripts/20200825095213_remove_orphaned_buffer_updates_by_article.rb:4]: Love that you went back and removed these so they wouldn't break for people cranking this up in the future! 🎉

Do we need to keep the files? I'm unfamiliar with the implementation we're using for data update scripts.

**Review by jgaskins — APPROVED** [review]
✂️✂️✂️

**Comment by benhalpern** [comment]
This is looking great.

Popping in to make sure this note from the issue doesn't get missed:

> Check with me before this is merged and deployed. We need to copy over a bit of buffer_updates data before we remove it.

"me" meaning me. 😅

File: lib/data_update_scripts/20200825095213_remove_orphaned_buffer_updates_by_article.rb (hunk 1 above)
Comment (citizen428) [reply lib/data_update_scripts/20200825095213_remove_orphaned_buffer_updates_by_article.rb:4]: Yeah, so often we do keep them around, but sometimes we just remove them. Good that you brought this up, I think I'm just going to nix them.

**Comment by citizen428** [comment]
> Don't know if you plan to remove them in the following PR

As much as I want to pretend that this was part of some grand design, I really just seem to be terrible at grep-ing.

**Review by rhymes — APPROVED** [review]
✂️

**Comment by benhalpern** [comment]
@citizen428 Thanks for this. I'm marking it as blocked until the social media team is 100% ready to make the switch.

**Review by benhalpern — APPROVED** [review]
Okay, we're good to go! ✂️

**Comment by citizen428** [comment]
I took this out of draft status and fixed the merge conflict, will merge once specs finish.

**Pull request merged**